armor62_signcrypt.go

```diff
@@ -19,26 +32,19 @@ import (
 //
 // The ciphertext is additionally armored with the recommended armor62-style format.
 //
-// Returns an io.WriteCloser that accepts plaintext data to be signcrypted; and
-// also returns an error if initialization failed.
-func NewSigncryptArmor62SealStream(ciphertext io.Writer, keyring Keyring, sender SigningSecretKey, receiverBoxKeys []BoxPublicKey, receiverSymmetricKeys []ReceiverSymmetricKey, brand string) (plaintext io.WriteCloser, err error) {
-	// Note: same "BEGIN SALTPACK ENCRYPTED" visible message type.
-	enc, err := NewArmor62EncoderStream(ciphertext, MessageTypeEncryption, brand)
-	if err != nil {
-		return nil, err
-	}
-	out, err := NewSigncryptSealStream(enc, keyring, sender, receiverBoxKeys, receiverSymmetricKeys)
-	if err != nil {
-		return nil, err
-	}
-	return closeForwarder([]io.WriteCloser{out, enc}), nil
+// If initialization succeeds, returns an io.WriteCloser that accepts
+// plaintext data to be signcrypted and a nil error. Otherwise,
+// returns nil and the initialization error.
+//
+// ephemeralKeyCreator should be the last argument; it's the 2nd one
+// to preserve the public API.
+func NewSigncryptArmor62SealStream(ciphertext io.Writer, ephemeralKeyCreator EphemeralKeyCreator, sender SigningSecretKey, receiverBoxKeys []BoxPublicKey, receiverSymmetricKeys []ReceiverSymmetricKey, brand string) (plaintext io.WriteCloser, err error) {
```

Comment:
changed `keyring` to `ephemeralKeyCreator`, but that's a backwards-compatible change, since `keyring` implements `ephemeralKeyCreator`